lib/bolt/logger.rb

```diff
@@ -123,18 +130,54 @@ def self.reset_logging
       Logging.reset
     end
 
-    def self.warn_once(type, msg)
+    # Issues a warning unless the user has suppressed it.
+    def self.warn(id:, msg:, src: self)
+      return if @disable_warnings.include?(id)
+
       @mutex.synchronize do
-        @logger ||= Bolt::Logger.logger(self)
-        if @warnings.add?(type)
-          @logger.warn(msg)
+        Bolt::Logger.logger(src).warn("#{msg} [ID: #{id}]")
+      end
+    end
+
+    # Issues a warning unless the user has suppressed it,
+    # or a warning with the same ID has already been issued.
+    def self.warn_once(id:, msg:, src: self)
```

Comment:
Adding in `src` fixes a (very minor) bug in the logger that shows all `warn_once` and `deprecation_warning` messages as coming from `Bolt::Logger` in a log file.